modules/bitgo/src/v2/coins/near.ts

```diff
@@ -269,13 +340,274 @@ export class Near extends BaseCoin {
       txHex: serializedTx,
     } as any;
   }
+  // TODO(BG-51092): Needs to be moved to a common place to re-use it for other coins
+  static async initializeMPCIfNotPresent() {
+    if (!this.initialized) {
+      const hdTree = await Ed25519BIP32.initialize();
+      this.MPC = await Eddsa.initialize(hdTree);
+      this.initialized = true;
+    }
+  }
+
+  // TODO(BG-51092): Needs to be moved to a common place to re-use it for other coins
+  async getTSSSignature(
+    userSigningMaterial: UserSigningMaterial,
+    backupSigningMaterial: BackupSigningMaterial,
+    path = 'm/0',
+    transaction: accountLib.Near.Transaction
+  ) {
+    await Near.initializeMPCIfNotPresent();
+
+    const userCombine = Near.MPC.keyCombine(userSigningMaterial.uShare, [
+      userSigningMaterial.bitgoYShare,
+      userSigningMaterial.backupYShare,
+    ]);
+    const backupCombine = Near.MPC.keyCombine(backupSigningMaterial.uShare, [
+      backupSigningMaterial.bitgoYShare,
+      backupSigningMaterial.userYShare,
+    ]);
+
+    const userSubkey = Near.MPC.keyDerive(
+      userSigningMaterial.uShare,
+      [userSigningMaterial.bitgoYShare, userSigningMaterial.backupYShare],
+      path
+    );
+
+    const backupSubkey = Near.MPC.keyCombine(backupSigningMaterial.uShare, [
+      userSubkey.yShares[2],
+      backupSigningMaterial.bitgoYShare,
+    ]);
+
+    const messageBuffer = transaction.signablePayload;
+    const userSignShare = Near.MPC.signShare(messageBuffer, userSubkey.pShare, [userCombine.jShares[2]]);
+    const backupSignShare = Near.MPC.signShare(messageBuffer, backupSubkey.pShare, [backupCombine.jShares[1]]);
+    const userSign = Near.MPC.sign(
+      messageBuffer,
+      userSignShare.xShare,
+      [backupSignShare.rShares[1]],
+      [userSigningMaterial.bitgoYShare]
+    );
+    const backup_sign = Near.MPC.sign(
```

Comment:
lets prefer consistent casing for variables

Reply:
Done